hooks/tk-multi-publish2/basic/publish_file.py

```diff
@@ -0,0 +1,341 @@
+
+        publisher = self.parent
+        path = item.properties.get("path")
+        is_sequence = item.properties.get("is_sequence", False)
+
+        if is_sequence:
+            # generate the name from one of the actual files in the sequence
+            name_path = item.properties["sequence_files"][0]
+        else:
+            name_path = path
+
+        # get the publish name for this file path. this will ensure we get a
+        # consistent publish name when looking up existing publishes.
+        publish_name = publisher.util.get_publish_name(
+            name_path, sequence=is_sequence)
+
+        self.logger.info("Publish name will be: %s" % (publish_name,))
+
+        # see if there are any other publishes of this path with a status.
+        # Note the name, context, and path *must* match the values supplied to
+        # register_publish in the publish phase in order for this to return an
+        # accurate list of previous publishes of this file.
+        publishes = publisher.util.get_conflicting_publishes(
+            item.context,
+            path,
+            publish_name,
+            filters=["sg_status_list", "is_not", None]
+        )
+
+        if publishes:
+            conflict_info = (
+                "If you continue, these conflicting publishes will no longer "
+                "be available to other users via the loader:<br>"
+                "<pre>%s</pre>" % (pprint.pformat(publishes),)
+            )
+            self.logger.warn(
+                "Found %s conflicting publishes in Shotgun" %
+                    (len(publishes),),
+                extra={
+                    "action_show_more_info": {
+                        "label": "Show Conflicts",
+                        "tooltip": "Show the conflicting publishes in Shotgun",
+                        "text": conflict_info
+                    }
+                }
+            )
+
+        return True
+
+    def publish(self, settings, item):
+        """
+        Executes the publish logic for the given item and settings.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+
+        publisher = self.parent
+        path = item.properties["path"]
+
+        # get the publish path components
+        path_info = publisher.util.get_file_path_components(path)
+
+        # determine the publish type
+        extension = path_info["extension"]
+        publish_type = self._get_publish_type(extension, settings)
+
+        is_sequence = item.properties.get("is_sequence", False)
+
+        if is_sequence:
+            # generate the name from one of the actual files in the sequence
+            name_path = item.properties["sequence_files"][0]
+        else:
+            name_path = path
+
+        # get the publish name for this file path. this will ensure we get a
```

Comment:
Seeing duplicated logic at the top of these methods. Could and should that be centralized?